Standards/scs-01xx-v1-dns.md

```diff
@@ -0,0 +1,122 @@
+
+## Motivation
+
+The behavior and feature set regarding DNS functionalities can differ greatly between OpenStack infrastructures, depending on their individual configuration.
+
+With this standard the SCS project aims to establish a baseline for reliable and consistent DNS features in SCS cloud environments.
+
+## Design Considerations
+
+OpenStack offers a lot of extension choices and even a dedicated service (Designate) for DNS functionality.
+The standard should make sure that a specified level of DNS functionality can be reached while taking into account that not all settings and choices might be feasible for CSPs to implement.
+
+### Options considered
+
+#### Making Designate mandatory
+
+To offer a consistent feature set to customers, the SCS project could consider to make Designate mandatory in a sense that SCS clouds would need to integrate the service, make it available to customers and properly configure it for publishing DNS records.
+This would offer easy DNS-as-a-Service functionality to customers.
+
+However, this would also require solid DNS expertise at CSP-side to properly set up and integrate Designate and DNS zones as Designate does not act as a full DNS server on its own but instead relies on external DNS providers or self-hosted DNS infrastructures that the CSP needs to integrate into it.
+
+#### Mandating the use of DNSSEC
+
+The DNSSEC extension to DNS ensures authenticity and integrity of the data provided to DNS resolvers.
+This protects against attacks like cache poisoning and increases the trustworthiness of the DNS infrastructure.
+
+This is a direct improvement for customers and yields low complexity for the CSP to implement since it is supported by most major DNS implementations natively.
+
+#### Mandating the use of local DNS recursors
+
+A local DNS recursor can be used to cache and serve DNS responses locally. It servers as a proxy between the clients and external DNS servers.
+This improves performance and speed of DNS resolution in the infrastructure.
+Furthermore, it can be configured to use DNSSEC, DNS over HTTPS and/or DNS over TLS to increase security and privacy of DNS requests it handles for clients.
+Even if individual clients do not support these functionalities, they can still benefit from a local recursor's implementation of those and don't have to trust external DNS servers directly.
+
+As such, the implementation of local DNS recursors in the infrastructure can be very beneficial.
+This standard should consider mandating or at least recommending the use of local DNS recursors for SCS clouds to be configured as the default DNS servers for Neutron resources.
+
+## Open questions
+
+RECOMMENDED
+
+## Decision
+
+The decisions of this standard will be categorized into three main sections: forwarded DNS, internal DNS and external DNS.
+
+Forwarded DNS refers to the DNS servers communicated to tenant VMs for DNS resolution as well as any local recursors involved.
```

Comment:
@horazont "Announced DNS"?

Reply:
An entirely new term where I wouldn't know when reading it what it actually means.

Maybe "Default DNS servers" is what describes it best?

Reply:
"forwarding" in DNS has a special defined meaning, please don't invent your own, to cite RFC 9499:

> Forwarding: The process of one server sending a DNS query with the RD bit set to 1 to another server to resolve that query. Forwarding is a function of a DNS resolver; it is different than simply blindly relaying queries.

> Forwarder: [Section 1](https://rfc-editor.org/rfc/rfc2308#section-1) of [[RFC2308](https://datatracker.ietf.org/doc/html/rfc2308)] describes a forwarder as "a nameserver used to resolve queries instead of directly using the authoritative nameserver chain". [[RFC2308](https://datatracker.ietf.org/doc/html/rfc2308)] further says "The forwarder typically either has better access to the internet, or maintains a bigger cache which may be shared amongst many resolvers." That definition appears to suggest that forwarders normally only query authoritative servers. In current use, however, forwarders often stand between stub resolvers and recursive servers. [[RFC2308](https://datatracker.ietf.org/doc/html/rfc2308)] is silent on whether a forwarder is iterative-only or can be a full-service resolver.

https://datatracker.ietf.org/doc/html/rfc9499#section-6-4.58

So please, do not invent your own terminology here, thanks.

Reply:
@artificial-intelligence Thank you for digging up the standard reference. This supports the claim that we need to use a term which is not related to "Forwarding".

(In OVN, there is no forwarder as er RFC9499 terminology.)

Reply:
> An entirely new term where I wouldn't know when reading it what it actually means.
> 
> Maybe "Default DNS servers" is what describes it best?

Since we seem to agree that "Forwarding" is a really misleading/incorrect choice here and no other suggestions have been made, I went with "Default DNS servers" for now. Feel free to comment other suggestions if any come to mind.